naxsi_src/naxsi_skeleton.c

```diff
@@ -2114,27 +2091,48 @@ ngx_http_naxsi_attack_action_variable(ngx_http_request_t*        r,
 }
 
 static ngx_int_t
-ngx_http_naxsi_request_id(ngx_http_request_t* r, ngx_http_variable_value_t* v, uintptr_t data)
+ngx_http_naxsi_request_id_variable(ngx_http_request_t*        r,
+                                   ngx_http_variable_value_t* v,
+                                   uintptr_t                  data)
 {
-  u_char*                 id  = NULL;
-  const size_t            len = NAXSI_REQUEST_ID_SIZE << 1;
-  ngx_http_request_ctx_t* ctx = ngx_http_get_module_ctx(r, ngx_http_naxsi_module);
+  char*   req_id = naxsi_request_id(r);
+  u_char* id     = NULL;
 
-  if (ctx == NULL) {
+  if (req_id == NULL) {
     return NGX_ERROR;
   }
 
-  id = ngx_pnalloc(r->pool, len);
+  id = ngx_pnalloc(r->pool, 32);
   if (id == NULL) {
     return NGX_ERROR;
   }
 
+  memcpy(id, req_id, 32);
+
   v->valid        = 1;
   v->no_cacheable = 0;
   v->not_found    = 0;
-  v->len          = len;
+  v->len          = 32;
   v->data         = id;
-
-  ngx_hex_dump(id, ctx->request_id, NAXSI_REQUEST_ID_SIZE);
   return NGX_OK;
 }
+
+char*
+naxsi_request_id(ngx_http_request_t* req)
+{
+  ngx_http_variable_value_t* lookup;
+
+  static ngx_str_t  request_id_varname  = ngx_string("request_id");
```

Comment:
This is only available from `NGINX 1.11.0` according to https://nginx.org/en/docs/http/ngx_http_core_module.html

Reply:
add a comment, so we can patch it when needed.

Reply:
also static vars are a problem, it should not be static at all.